opal/mca/btl/tcp/btl_tcp.c

```diff
@@ -552,9 +552,12 @@ int mca_btl_tcp_recv_blocking(int sd, void* data, size_t size)
     while (cnt < size) {
         int retval = recv(sd, ((char *)ptr) + cnt, size - cnt, 0);
         /* remote closed connection */
-        if (0 == retval) {
+        if (0 == retval && 0 == cnt) {
+            /* nothing was read at all, this might be find, so simply returns */
+            return 0;
+        } else if (0 == retval) {
             BTL_ERROR(("remote peer unexpectedly closed connection while I was waiting for blocking message"));
-            return -1;
+            return cnt;
         }
```

Comment:
@ggouaillardet ya you are right if we read "something but not all" then we throw error. In case of cnt being 0 ( where here both the connection are from MPI proc but happens simultaneously) then one of the connection will be dropped because of error. But if we read in between 0 - expected size then we are throwing error as we expect (guid and magic string) to recognize connection is from MPI proc.